queries.sql

```diff
@@ -9,3 +9,37 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name NOT IN ('Gabumon');
 SELECT * FROM animals WHERE  weight_kg BETWEEN 10.4 and 17.3;
+
+SELECT * FROM animals;
+BEGIN TRANSACTION;
+UPDATE animals SET species='unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+SELECT * FROM animals;
+COMMIT;
+SELECT * FROM animals;
+BEGIN TRANSACTION;
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+BEGIN TRANSACTION;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT DELDOB;
+UPDATE animals SET weight_kg = weight_kg * -1;
+SELECT * FROM animals;
+ROLLBACK TO DELDOB;
+SELECT * FROM animals;
```

Comment:
- [ ] Just as I mentioned above, I'll suggest the statement `BEGIN TRANSACTION;` comes first in all these transactions before any other line of code in the transaction. :+1: :+1:

Reply:
-Refactored code to resolve bug